### PR TITLE
Upgrade `network` package to `2021-02-01`

### LINKED ISSUE #12983: Support new resource of  `Custom IP prefix`
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I will support the new resource network/custom IP prefix in TF. The network API version currently used by TF is [v2020-11-01](https://github.com/Azure/azure-rest-api-specs/tree/master/specification/network/resource-manager/Microsoft.Network/stable/2020-11-01).

Because I need to use the new parameters added in[ the latest version of the custom ip prefix API](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2021-02-01/customIpPrefix.json), should upgrade the network API to the latest version [v2021-02-01](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2021-02-01) in TF before adding custom IP prefix.

Could you help me to upgrade network API to [the latest version](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2021-02-01) in TF?

### New or Affected Resource(s)

* azurerm_custom_ip_prefix

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

### PR DESCRIPTION
**Pull request by favoretti**
Fixes #12983

### CODE REVIEW

**Comment by favoretti** [comment]
LB tests pass:
image

**Comment by favoretti** [comment]
FW tests pass:
image

**Comment by favoretti** [comment]
Network tests pass (failures are unrelated or not relevant):
image

File: internal/services/network/virtual_network_peering_resource.go [hunk 1]
```diff
@@ -226,7 +226,7 @@ func retryVnetPeeringsClientCreateUpdate(d *pluginsdk.ResourceData, resGroup str
 		ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
 		defer cancel()
 
-		future, err := vnetPeeringsClient.CreateOrUpdate(ctx, resGroup, vnetName, name, peer)
+		future, err := vnetPeeringsClient.CreateOrUpdate(ctx, resGroup, vnetName, name, peer, network.SyncRemoteAddressSpaceTrue)
```
Comment (favoretti) [inline internal/services/network/virtual_network_peering_resource.go:229]: This is the only available option for now, so decided not to make it into a parameter for now.

**Review by katbyte — APPROVED** [review]
Thanks @favoretti - LGTM 🏗️

**Pull request merged**